src/gdal/ogr_file_format.cpp

```diff
@@ -502,7 +502,7 @@ namespace {
 	 * 
 	 * \see CutoutOperation::operator()
 	 */
-	class ClippingImplementation : public OgrFileImport::Clipping
+	class ClippingImplementation final : public OgrFileImport::Clipping
```

Comment:
I saw this warning just yesterday in Qt Creator (static analyzer), but didn't stop for investigation. Is this an error without `-Werror`? `final` on class *and* destructor seems redundant. (Same for class `Progress`).

Reply:
It is merely a warning. I took the liberty to eliminate most of the diagnostic messages that clang spit out. Moving the keyword from the destructor to the class level seems appropriate to me.